nixos/modules/services/audio/wyoming/piper.nix

```
@@ -0,0 +1,172 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+let
+  cfg = config.services.wyoming.piper;
+
+  inherit (lib)
+    escapeShellArgs
+    literalExpression
+    mkOption
+    mdDoc
+    mkEnableOption
+    mkPackageOptionMD
+    types
+  ;
+
+in
+
+{
+  meta.buildDocsInSandbox = false;
+
+  options.services.wymonig.piper = with types; {
+    package = mkPackageOptionMD pkgs "wyoming-piper" { };
+
+    servers = mkOption {
+      default = {};
+      description = mdDoc ''
+        Attribute set of piper instances to spawn.
+      '';
+      type = types.attrsOf (types.submodule (
+        { ... }: {
+          options = {
+            enable = mkEnableOption (mdDoc "Wyoming Piper server");
+
+            piper = mkOption {
+              type = path;
+              default = "${pkgs.piper-tts}/bin/piper";
+              example = literalExpression "''${pkgs.piper-tts}/bin/piper";
+              description = mdDoc ''
+                Path to the piper executable
+              '';
+            };
+
+            voice = mkOption {
+              type = str;
+              example = "en-us-ryan-medium";
+              description = mdDoc ''
+                Name of the voice model to use. See the following website for samples:
+                https://rhasspy.github.io/piper-samples/
+              '';
+            };
+
+            uri = mkOption {
+              type = strMatching "^(tcp|unix)://.*$";
+              example = "tcp://0.0.0.0:10200";
+              description = mdDoc ''
+                URI to bind the wyoming server to.
+              '';
+            };
+
+            speaker = mkOption {
+              type = ints.positive;
```

Should be the following as 0 is valid.
```suggestion
              type = ints.unsigned;
```